src/core/structure/sheet.ts

```diff
@@ -435,46 +450,161 @@ export default class Sheet {
+      this.emitSetStyleEvent(
+        columnIndex,
+        rowIndex,
+        GenerateStyleStringFromMap(this.getMergedCellStyle(columnKey).styling),
+      );
+      return;
+    }
+
+    column.cellFormatting.set(
+      row.key,
+      new CellStyle(css, column.cellFormatting.get(rowKey!)?.formatter),
+    );
+    row.cellFormatting.set(
+      column.key,
+      new CellStyle(css, row.cellFormatting.get(columnKey!)?.formatter),
+    );
+
+    this.emitSetStyleEvent(
+      columnIndex,
+      rowIndex,
+      GenerateStyleStringFromMap(
+        this.getMergedCellStyle(columnKey, rowKey).styling,
+      ),
+    );
   }
 
-  setRowStyle(rowKey: RowKey, style: CellStyle | null): boolean {
-    const row = this.rows.get(rowKey);
-    if (!row) return false;
+  setGroupCss(
+    groupIndex: number,
+    groupType: GroupType,
+    css: Map<string, string> = new Map(),
+  ): void {
+    const isColumnGroup = GroupTypes.Column == groupType;
+    const positions = isColumnGroup ? this.columnPositions : this.rowPositions;
+    const groupKey = isColumnGroup
+      ? (positions.get(groupIndex!) as ColumnKey)
+      : (positions.get(groupIndex) as RowKey);
+    if (!groupKey) return;
+    const group = isColumnGroup
+      ? this.columns.get(groupKey as ColumnKey)
+      : this.rows.get(groupKey as RowKey);
+    if (!group) return;
+
+    if (!css || css.size == 0) {
+      group.defaultStyle?.clearCss();
+      this.emitSetStyleEvent(
+        isColumnGroup ? groupIndex : null,
+        !isColumnGroup ? groupIndex : null,
+        GenerateStyleStringFromMap(
+          isColumnGroup
+            ? this.getMergedCellStyle(groupKey as ColumnKey).styling
+            : this.getMergedCellStyle(null, groupKey as RowKey).styling,
+        ),
+      );
+      return;
+    }
 
-    this.setCellGroupStyle(row, style);
-    return true;
+    group.defaultStyle = new CellStyle(css, group.defaultStyle?.formatter);
+
+    this.emitSetStyleEvent(
+      isColumnGroup ? groupIndex : null,
+      !isColumnGroup ? groupIndex : null,
+      GenerateStyleStringFromMap(
+        isColumnGroup
+          ? this.getMergedCellStyle(groupKey as ColumnKey).styling
+          : this.getMergedCellStyle(null, groupKey as RowKey).styling,
+      ),
+    );
+  }
+
+  setGroupFormatter(
+    groupIndex: number,
+    groupType: GroupType,
+    formatter: Formatter | null = null,
+  ): void {
+    const isColumnGroup: boolean = groupType == GroupTypes.Column;
+    const groupKey = isColumnGroup
+      ? this.columnPositions.get(groupIndex)
+      : this.rowPositions.get(groupIndex);
+    if (!groupKey) return;
+    const group = isColumnGroup
+      ? this.columns.get(groupKey as ColumnKey)
+      : this.rows.get(groupKey as RowKey);
+    if (!group) return;
+
+    const cellStyle = group.defaultStyle?.styling
+      ? new CellStyle(group.defaultStyle?.styling, formatter)
+      : new CellStyle(null, formatter);
+
+    this.setCellGroupStyle(group, cellStyle);
   }
```

Comment:
"Group" should be an internal concept only. We shouldn't introduce `GroupType` etc. These methods are also filled with group type checks, maybe just split into row/column or refactor to match other `CellGroup` methods.